store/badger.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2019 Perlin
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of
+// this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to
+// use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+package store
+
+import (
+	badger "github.com/dgraph-io/badger"
+	"github.com/pkg/errors"
+)
+
+type badgerWriteBatch struct {
+	batch *badger.WriteBatch
+}
+
+func (b *badgerWriteBatch) Put(key, value []byte) {
+	k := append([]byte{}, key...)
+	v := append([]byte{}, value...)
+	err := b.batch.Set(k, v)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Not sure if this is the best way to handle the error returned by Set()

It seems that if the batch is too big, the Set() will call commit() internally. So, the error might be coming from commit().

Reply:
Good catch. Looks like `WriteBatch.Put` needs to return an error. And it must be made clear that `WriteBatch` does not guarantee all of the batched operations are not committed if it's not explicitly committed, as the underlying database may choose to commit the write batch earlier.